[systemctl-user] Fix systemctl-user edit command. Fixes JB#56246

Tomin1 opened the pull request:
Previously running systemctl-user edit as root opened the file in root's home directory. Setting HOME to seat0 user opens the expected file. Set also USER environment variable for good measure.

Make systemctl replace shell. If systemctl-user is used as part of systemd unit it might not be properly terminated sometimes. This makes systemctl to replace shell which allows systemd to properly manage process lifetimes.

Thaodan commented:
The change in the submodule is intended?

Tomin1 commented:
> The change in the submodule is intended?

What change in submodule? This should change only systemctl-user script and at least I don't see any other change here.

Thaodan commented:
> > The change in the submodule is intended?
> 
> What change in submodule? This should change only systemctl-user script and
> at least I don't see any other change here.
> 

https://github.com/sailfishos/systemd/pull/6/commits/
172e3eef4026fef7f0b0d0d506dd7675ecda2d53

Github showed one in the commit above, I guess that's wrong.

Tomin1 commented:
> https://github.com/sailfishos/systemd/pull/6/commits/ [172e3ee](https://github.com/sailfishos/systemd/commit/172e3eef4026fef7f0b0d0d506dd7675ecda2d53) Github showed one in the commit above, I guess that's wrong.

Yeah, there was a mistake. I fixed it now and rebased too.

rainemak reviewed (approved):
LGTM

rainemak commented:
Please merge when you have time. Also remember to delete the branch. :+1:
